india_compliance/gst_india/doctype/gst_return_log/generate_gstr_1.py

```diff
@@ -747,16 +747,22 @@ def process_reset_gstr1(self):
 
         return response
 
-    def upload_gstr1(self, json_data, force):
+    def upload_gstr1(self, json_data, is_nil_rated, force):
         if not json_data:
             return
 
         verify_request_in_progress(self, force)
 
         keys = {category.value for category in GovJsonKey}
         if all(key not in json_data for key in keys):
-            frappe.msgprint(_("No data to upload"), indicator="red")
-            return
+            if not cint(is_nil_rated):
+                frappe.msgprint(
+                    _("No data to upload. To file Nil Return, mark the checkbox."),
```

Comment:
Error message: No data to upload.To file Nil Return Select "File Nil GSTR-1" checkbox.